### PR TITLE
Initial Funneler gem as extracted from Doximity

### PR DESCRIPTION
**Pull request by rstawarz**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+source 'https://rubygems.org'
+
```
Comment (chriswoodrich) [inline Gemfile]: Why is the gemfile empty if Gemfile.lock has entries?

File: bin/console [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env ruby
```
Comment (chriswoodrich) [inline bin/console]: cool

File: funneler.gemspec [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+Gem::Specification.new do |spec|
+  spec.name          = "funneler"
+  spec.version       = Funneler::VERSION
+  spec.authors       = ["Ryan Stawarz"]
+  spec.email         = ["ryan@stawarz.com"]
+
+  spec.summary       = %q{A light-weight approach for routing users through a pre-determined set of routes.}
+  spec.description   = %q{}
+  spec.homepage      = ""
```
Comment (chriswoodrich) [inline funneler.gemspec]: might as well plug your main github page here

File: Gemfile (hunk 1 above)
Comment (rstawarz) [inline Gemfile]: when you bundle install, it also looks at the gem spec, and there are some dependencies in there

File: README.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
+# Funneler
+
+A simple gem for helping coordinate moving a user through a set of
+routes.  The gem uses a token to keep track of the current place in the funnel.
+The simplest way to use that token is by including a module into the controller
+and pass the token in the route. However since the token is simply a string
```
Comment (brupm) [inline README.md]: passing

File: README.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+## Development
+
+After checking out the repo, run `bin/setup` to install dependencies. Then, run `rake spec` to run the tests. You can also run `bin/console` for an interactive prompt that will allow you to experiment.
+
+To install this gem onto your local machine, run `bundle exec rake install`. To release a new version, update the version number in `version.rb`, and then run `bundle exec rake release`, which will create a git tag for the version, push git commits and tags, and push the `.gem` file to [rubygems.org](https://rubygems.org).
+
+## Contributing
+
+Bug reports and pull requests are welcome on GitHub at https://github.com/doximity/funneler. This project is intended to be a safe, welcoming space for collaboration, and contributors are expected to adhere to the [Contributor Covenant](http://contributor-covenant.org) code of conduct.
```
Comment (brupm) [inline README.md]: welcomed

File: lib/funneler/version.rb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+module Funneler
+  VERSION = "1.0"
```
Comment (rstawarz) [inline lib/funneler/version.rb]: @fgrehm - This gem has been incubated within Doximity app for about 3mo now and is fairly stable.  You can see its extraction here: https://github.com/doximity/doximity/issues/16845

Comment (fgrehm) [inline lib/funneler/version.rb]: Yeah, but I think we need an extra version number `.0` to conform to semver and allow us to cut "patch releases" :smile:

Comment (rstawarz) [inline lib/funneler/version.rb]: ah.. i see what you're saying, my bad!  I think that's inferred, but I'll update that to be certain

**Pull request merged**